### PR TITLE
Verify config.json and add all the missing info

### PR DESCRIPTION
**Pull request by agherzan**
Verify config.json and add all the missing info there based on running flag

Based on --staging argument (default to production when not present),
config.json will be check for all the required attributes and fix accordingly.

Signed-off-by: Andrei Gherzan <andrei@resin.io>

CC @petrosagg @telphan @floion

PR for #3

[Review on Reviewable](https://reviewable.io/reviews/resin-os/resinhup/4)

### CODE REVIEW

File: conf/resinhup.conf [hunk 1]
```diff
@@ -45,3 +45,23 @@ to_keep_files:
 remote: https://s3.amazonaws.com/resin-production-hup/
 workspace: /tmp/resinhup
 update_file_fingerprints: ./resin-boot ./resin-root
+
+[production]
```
Comment (petrosagg) [inline conf/resinhup.conf]: The labels are the other way around

Comment (agherzan) [inline conf/resinhup.conf]: @petrosagg fixed. Thanks for this.

File: modules/util.py [hunk 2]
```diff
@@ -350,3 +377,26 @@ def mcopy(dev, src, dst):
         log.debug("Failed to mcopy in " + dev);
         return False
     return True
+
+def jsonAttributeExists(json, attribute):
```
Comment (petrosagg) [inline modules/util.py]: Why are we using `jq` for reading and writing JSON? https://docs.python.org/3/library/json.html

Comment (agherzan) [inline modules/util.py]: :) I will use the library. I thought you wanted to use bash commands as much as possible.

Comment (agherzan) [inline modules/util.py]: Implemented in the new PR version. Thanks.

File: modules/util.py [hunk 3]
```diff
@@ -350,3 +378,46 @@ def mcopy(dev, src, dst):
...
+            return False
+        else:
+            log.warn("jsonSetAttribute: " + attribute + " will be overwritten.")
+
+    configjson[atttribute] = value # Set the required attribute
+
+    # Write new json to a tmp file
+    try:
+        with open(json+'.hup.tmp', 'w') as fd:
+            configjson = json.dump(configjson, fd)
```
Comment (petrosagg) [inline modules/util.py]: missing `fsync` call

Comment (agherzan) [inline modules/util.py]: Added. Thanks.

**Comment by petrosagg** [comment]
SGTM

**Pull request merged**